src/md_lammps.c

```diff
@@ -473,4 +474,527 @@ static md_molecule_loader_i lammps_api = {
+	ASSERT(reader);
+	ASSERT(alloc);
+	str_t line;
+	str_t tokens[6];
+	int64_t num_frames = 0;
+
+
+	//Setup frames and first frame
+	md_buffered_reader_skip_line(reader);
+	md_buffered_reader_extract_line(&line, reader);
+	md_array_push(*frame_times, parse_int(line), alloc);
+	num_frames++;
+
+
+	//Parse num of atoms
+	{
+		md_buffered_reader_skip_line(reader);
+		md_buffered_reader_extract_line(&line, reader); //Read num of atoms
+		*num_atoms = parse_int(line);
+	}
+
+	//Parse unit_cell definition
+	{
+		float cell_extent[3];
+		float cell_tri[3]; //Triclinic
+
+		md_buffered_reader_extract_line(&line, reader); //Read BOX BOUNDS
+		if (str_equal_cstr(line, "ITEM: BOX BOUNDS pp pp pp")) {
+			//Cubic
+			for (int8_t i = 0; i < 3; i++) {
+				md_buffered_reader_extract_line(&line, reader);
+				extract_tokens(tokens, 4, &line);
+				cell_extent[i] = (float)(parse_float(tokens[1]) - parse_float(tokens[0]));
+			}
+			*unit_cell = md_util_unit_cell_from_extent(cell_extent[0], cell_extent[1], cell_extent[2]);
+		}
+		else if (str_equal_cstr(line, "ITEM: BOX BOUNDS xy xz yz pp pp pp")) {
+			//Triclinic
+			for (int8_t i = 0; i < 3; i++) {
+				md_buffered_reader_extract_line(&line, reader);
+				extract_tokens(tokens, 4, &line);
+				cell_extent[i] = (float)(parse_float(tokens[1]) - parse_float(tokens[0]));
+				cell_tri[i] = (float)parse_float(tokens[2]);
+			}
+			*unit_cell = md_util_unit_cell_from_triclinic(cell_extent[0], cell_extent[1], cell_extent[2], cell_tri[0], cell_tri[1], cell_tri[2]);
+		}
+		else {
+			MD_LOG_ERROR("Could not correctly parse BOX BOUND");
+			return false;
+		}
+	}
+
+	while (md_buffered_reader_extract_line(&line, reader)) {
+
+		if (str_equal_cstr_n(line, "ITEM: TIMESTEP", 14)) {
+			md_buffered_reader_extract_line(&line, reader);
+			num_frames++;
+			md_array_push(*frame_times, parse_int(line), alloc);
+		}
+		else if (str_equal_cstr_n(line, "ITEM: ATOMS", 11)) {
+
+			// This is a bit nasty, we want to get the correct offset to the beginning of the current line.
+			// Therefore we need to do some pointer arithmetic because just using the length of the line may not get us
+			// all the way back in case there were skipped \r characters.
+			const int64_t offset = md_buffered_reader_tellg(reader) - (reader->str.ptr - line.ptr);
+			md_array_push(*frame_offsets, offset, alloc);
+		}
+	}
+
+	//We add the end of the file to frame_offsets, so frame_offset size = num_frames + 1
+
+	const int64_t end_of_file = md_buffered_reader_tellg(reader);
+	md_array_push(*frame_offsets, end_of_file, alloc);
+	*num_frame_offsets = (int64_t)md_array_size(*frame_offsets);
+
+	return true;
+}
+
+bool md_lammps_trajectory_parse_file(int64_t* num_atoms, md_unit_cell_t* unit_cell, int64_t* num_frame_offsets, md_array(int64_t)* frame_offsets, md_array(int64_t)* frame_times, str_t filename, struct md_allocator_i* alloc) {
```

Comment:
make this static and remove the declaration from the header as it should not be exposed.

Reply:
Made static and removd from .h